### PR TITLE
Move simulation/run script generation to separate methods for `Questa`

### PR DESCRIPTION
**Pull request by benbr8**
This makes it easier to implement custom extension of `Questa` class, for example for use case #169.

### CODE REVIEW

**Comment by themperek** [comment]
@benbr8 Sorry but this will need to rebase now. I needed to do some cleanup.

**Comment by themperek** [comment]
I have promised to do a release now but we can make another one asap if you need it.

**Comment by benbr8** [comment]
I have rebased to the current master. Don't worry about the release, I do not rely on PyPI to get the package :).

**Comment by themperek** [comment]
I am sorry for this but one more option is probably to rewrite this part like this: https://github.com/cocotb/cocotb/blob/d56b76575f81185493cf5ed705eae7621869e316/cocotb/runner.py#L439-L444 (and bellow).
It is a bit nicer and expects easier for the user. Wat do you think?

**Comment by benbr8** [comment]
I need to be able to input custom commands into `do_script` before `run -all; quit`. One solution could be to create a method which provides the initial value for `do_script` which can be overloaded by a custom implementation.

Let me see if I can come up with something.

**Comment by themperek** [comment]
> 

Yes. But if we follow `cocotb/runner.py` it gets much simpler.

File: cocotb_test/simulator.py [hunk 1]
```diff
@@ -491,40 +491,20 @@ def get_define_commands(self, defines):
...
+            do_script += "log -recursive /*;"
+        return do_script
 
     def build_command(self):
 
         cmd = []
 
+        do_script = self.do_script()
+        if not self.gui:
+            do_script += "run -all; quit"
```
Comment (themperek) [inline cocotb_test/simulator.py:506]: Would add this also to `do_script`.  Can imagine someone wants to stop in between and execute some command (like save/restore simulation state)

Opinion?

Comment (benbr8) [reply cocotb_test/simulator.py:506]: True, users then can prepend their custom scripts like this
```python
class QuestaCustom(Questa):
    def do_script(self):
        do_script = f"coverage save -onexit {self.module}.ucdb;"
        do_script += super().do_script()
        return do_script
```

**Review by themperek — APPROVED** [review]

File: cocotb_test/simulator.py (hunk 1 above)
Comment (themperek) [reply cocotb_test/simulator.py]: 👍🏻

**Pull request merged**